docs/react/migration-v4.en-US.md

```diff
@@ -3,31 +3,151 @@ order: 8
+- Mention removed, use `Mentions` instead.
+- Removed the `iconType` property of Alert. Please use `icon` instead.
+- Removed the `iconType` attribute of Modal.xxx. Please use `icon` instead.
+- Removed the Form.create method, `form` is now available in `Form.useForm`.
+- Removed the `id` attribute of Form.Item. Please use `htmlFor` instead.
+- The `setContentRef` property of Typography has been removed, please use `ref` instead.
+- Removed the `allowEmpty` property of TimePicker, please use `allowClear` instead.
+- Removed `AfterClose` attribute of Tag, please use `OnClose` instead.
+- Removed the `noHovering` property of Card, please use `hoverable` instead.
+- Removed the `vertical` property of Carousel. Please use `dotPosition` instead.
+- Removed `wrapClassName` property of Drawer, please use `className` instead.
+- Removed the `autosize` property of TextArea. Please use `autoSize` instead.
+- Removed the `offset` attribute of Affix. Please use `offsetTop` instead.
+- Removed the `onSearchChange` property of Transfer. Please use `onSearch` instead.
+- Removed the `body` attribute of Transfer. Please use `children` instead.
+- Removed the `lazy` attribute of Transfer, which did not really optimize the effect.
+- Removed `combobox` mode, please use `AutoComplete` instead.
+
+#### Icon upgrade
+
+In `antd @ 3.9.0`, we introduced the svg icon ([Why use the svg icon?](Https://github.com/ant-design/ant-design/issues/10353)). The icon API using the string name cannot be loaded on demand, so the svg icon file is fully introduced, which greatly increases the size of the packaged product. In 4.0, we adjusted the icon usage API to support tree shaking, reducing the default package size of antd by about 150 KB (Gzipped).
+
+Legacy Icon usage will be discarded:
+
+```jsx
+import { Icon, Button } from 'antd';
+
+const Demo = () => (
+  <div>
+    <Icon type="smile" />
+    <Button icon="smile" />
+  </div>
+);
+```
+
+It will be imported on demand in v4:
+
+```diff
+import { Button } from 'antd';
+
+// tree-shaking supported
+- import { Icon } from 'antd';
++ import { SmileOutlined } from '@ant-design/icons';
+
+const Demo = () => (
+  <div>
+-    <Icon type="smile" />
++    <SmileOutlined />
+    <Button icon={<SmileOutlined />} />
+  </div>
+);
+
+// or directly import
+import SmileOutlined from '@ant-design/icons/SmileOutlined';
+```
+
+You will still be able to continue using the compatibility pack:
+
+```jsx
+import { Button } from 'antd';
+import { Icon } from '@ant-design/compatible';
+
+const Demo = () => (
+  <div>
+    <Icon type="smile" />
+    <Button icon="smile" />
+  </div>
+);
+```
+
+#### Component refactoring
+
+- Form rewrite. No need to use `Form.create`. See [here](/components/form/v3) for migration documentation.
+- DatePicker rewrite
+  - Provide the `picker` property for selector switching.
+  - Range selection can now select start and end times individually.
+- Tree, Select, TreeSelect, AutoComplete use virtual scrolling.
```

Comment:
not Table ?

Reply:
Migration document only list breaking change. changelog list fully update.